cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaMirrorMaker2AssemblyOperatorTest.java

```diff
@@ -644,4 +644,86 @@ public void testCreateClusterStatusNotReady(VertxTestContext context) {
+        when(mockPdbOps.reconcile(anyString(), any(), pdbCaptor.capture())).thenReturn(Future.succeededFuture());
+
+        ArgumentCaptor<KafkaMirrorMaker2> mirrorMaker2Captor = ArgumentCaptor.forClass(KafkaMirrorMaker2.class);
+        when(mockMirrorMaker2Ops.updateStatusAsync(mirrorMaker2Captor.capture())).thenReturn(Future.succeededFuture());
+
+        KafkaConnectApi mockConnectClient = mock(KafkaConnectApi.class);
+        when(mockConnectClient.list(anyString(), anyInt())).thenReturn(Future.succeededFuture(emptyList()));
+
+        KafkaMirrorMaker2AssemblyOperator ops = new KafkaMirrorMaker2AssemblyOperator(vertx, new PlatformFeaturesAvailability(true, kubernetesVersion),
+                supplier, ResourceUtils.dummyClusterOperatorConfig(VERSIONS), x -> mockConnectClient);
+
+        KafkaMirrorMaker2Cluster mirrorMaker2 = KafkaMirrorMaker2Cluster.fromCrd(kmm2, VERSIONS);
+
+        Checkpoint async = context.checkpoint();
+        ops.createOrUpdate(new Reconciliation("test-trigger", KafkaMirrorMaker2.RESOURCE_KIND, kmm2Namespace, kmm2Name), kmm2)
+                .onComplete(context.succeeding(v -> context.verify(() -> {
+
+                    // Verify service
+                    List<Service> capturedServices = serviceCaptor.getAllValues();
+                    assertThat(capturedServices, hasSize(1));
+                    Service service = capturedServices.get(0);
+                    assertThat(service.getMetadata().getName(), is(mirrorMaker2.getServiceName()));
+                    assertThat(service, is(mirrorMaker2.generateService()));
+
+                    // Verify Deployment
+                    List<Deployment> capturedDc = dcCaptor.getAllValues();
+                    assertThat(capturedDc, hasSize(1));
+                    Deployment dc = capturedDc.get(0);
+                    assertThat(dc.getMetadata().getName(), is(mirrorMaker2.getName()));
+                    Map<String, String> annotations = new HashMap<>();
+                    annotations.put(Annotations.STRIMZI_LOGGING_ANNOTATION, LOGGING_CONFIG);
+                    assertThat(dc, is(mirrorMaker2.generateDeployment(annotations, true, null, null)));
+
+                    // Verify PodDisruptionBudget
+                    List<PodDisruptionBudget> capturedPdb = pdbCaptor.getAllValues();
+                    assertThat(capturedPdb, hasSize(1));
+                    PodDisruptionBudget pdb = capturedPdb.get(0);
+                    assertThat(pdb.getMetadata().getName(), is(mirrorMaker2.getName()));
+                    assertThat(pdb, is(mirrorMaker2.generatePodDisruptionBudget()));
+
+                    // Verify status
+                    List<KafkaMirrorMaker2> capturedMirrorMaker2s = mirrorMaker2Captor.getAllValues();
+                    assertThat(capturedMirrorMaker2s.get(0).getStatus().getUrl(), is("http://foo-mirrormaker2-api.test.svc:8083"));
```

Comment:
As with most of the other resources - can we make sure that the URL is listed only when it can be used? (i.e. when replicas > 0)?